models/schemas/userSchema.ts

```diff
@@ -41,8 +42,24 @@ const UserSchema = new mongoose.Schema({
   },
 });
 
+const generateSlug = function(username:string, level_name:string):string {
+  return username +
+  '/' +
+  level_name.replace(/\s+/g, '-').toLowerCase();
+};
+UserSchema.post('updateOne', async function() {
+  // if name has changed then call save on every level belonging to the user
+  if (this.getUpdate().$set.name) {
+    const levels = await LevelModel.find({
```

Comment:
Would an updateMany be faster here? Although not sure how easy it is / if it's possible to use the generateSlug function within an update query. If it's annoying to add I'm fine with this

Reply:
I tried with updateMany for a while... Unfortunately I couldn't get the updateMany hook to work in LevelSchema. Maybe there is a way to have Mongo itself do the regex code to generate the slug and reference a documents own name field... I can look into it as a follow up or I can try again if you would prefer

Reply:
I think it's fine to look into as a follow up. Username changes aren't that frequent anyways

Reply:
Just did some trial and error. Apparently the newer versions of mongo (which i am assuming we are using) support using some manipulation operators in update queries (so we can reference other fields). However, i can't find any examples using regex replace. looks like only basic replace is supported (or basic things like concat)